### PR TITLE
Replace Invariant container types with wider types where applicable.

### PR DESCRIPTION
**Pull request by bijij**
## Summary

This PR replaces Invariant type annotations (List/Dict) with wider ABCs where applicable.

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/emoji.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ async def delete(self, *, reason: Optional[str] = None) -> None:
 
         await self._state.http.delete_custom_emoji(self.guild_id, self.id, reason=reason)
 
-    async def edit(self, *, name: str = MISSING, roles: List[Snowflake] = MISSING, reason: Optional[str] = None) -> Emoji:
+    async def edit(self, *, name: str = MISSING, roles: Collection[Snowflake] = MISSING, reason: Optional[str] = None) -> Emoji:
```
Comment (bijij) [inline discord/emoji.py:217]: This could be `Iterable` if we want.

File: discord/guild.py [hunk 2]
```diff
@@ -1063,7 +1065,7 @@ def _create_channel(
         self,
         name: str,
         channel_type: Literal[ChannelType.text],
-        overwrites: Dict[Union[Role, Member], PermissionOverwrite] = ...,
+        overwrites: Mapping[Union[Role, Member], PermissionOverwrite] = ...,
```
Comment (bijij) [inline discord/guild.py:1068]: I'm not sure How I feel about these `Mapping` replacements.
We used to do a `isinstance(x, dict)` check which had to be modified to support it.

Undo?

**Review by bijij — COMMENTED** [review]
Most of these changes are just List>Sequence in the send/edit methods, I've commented on the cases I wasn't completely sure on.

File: discord/guild.py (hunk 2 above)
Comment (Rapptz) [reply discord/guild.py:1068]: I can't readily imagine someone using a non-dict `Mapping` being used here.

**Pull request merged**